### PR TITLE
Add Lisk support

### PR DESCRIPTION
**Pull request by alepop**
This is a work in progress to add support for `Lisk` to the `trezor-mcu`.
Messages: https://github.com/trezor/trezor-common/pull/68
Trezor-pyton: https://github.com/trezor/python-trezor/pull/197

Progress: 
- [x] Lisk get address
- [x] Lisk get public key
- [x] Lisk sign message
- [x] Lisk verify message
- [x] Lisk sign tx
  - [x] send transaction
  - [x] signature ransaction
  - [x] delegate transaction
  - [x] vote transaction
  - [x] multisignature transaction

### CODE REVIEW

**Comment by alepop** [comment]
@prusnak I'm not good at `C `language and maybe I did a lot of newbie mistakes. Please count on it.  All test works locally and emulator code compiled fine without errors but on CI it falls. (for especially for `sscanf` and  `sprintf`)

**Comment by prusnak** [comment]
Can you please rebase and squash your commits on top of the current master?

**Comment by alepop** [comment]
@prusnak done

**Comment by prusnak** [comment]
Thanks!

**Comment by prusnak** [comment]
Will address some of the issues in subsequent commit.

**Comment by prusnak** [comment]
@alepop I did some improvements in the master branch. Please test whether everything works like it should. The tests pass, but I might have broken something in the UI, etc.

**Comment by alepop** [comment]
@prusnak everything is good 👍

File: firmware/fsm_msg_lisk.h [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    CHECK_PIN
+
+    RESP_INIT(LiskMessageSignature);
+
+    HDNode *node = fsm_getDerivedNode(ED25519_NAME, msg->address_n, msg->address_n_count, NULL);
+    if (!node) return;
+
+    hdnode_fill_public_key(node);
+
+    lisk_sign_message(node, msg, resp);
```
Comment (keepkeyjon) [inline firmware/fsm_msg_lisk.h]: Slightly strange behavior here compared to the rest of the Trezor codebase: if the user cancels signing, you get a Failure::ActionCancelled, as well as a LiskMessageSignature without the signature populated. Most everywhere else, you'd only get the failure after the user cancels.

IMO lisk_sign_message shouldn't be touching fsm, and instead that confirmation / cancelling should happen here.

File: firmware/fsm_msg_lisk.h [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	CHECK_PIN
+
+    RESP_INIT(LiskSignedTx);
+
+    HDNode *node = fsm_getDerivedNode(ED25519_NAME, msg->address_n, msg->address_n_count, NULL);
+    if (!node) return;
+
+    hdnode_fill_public_key(node);
+
+    lisk_sign_tx(node, msg, resp);
```
Comment (keepkeyjon) [inline firmware/fsm_msg_lisk.h]: Likewise, if the user cancels during signing here, you'll still get both messages.

**Pull request merged**